test/unit/spec/util/jobLogViewerSpec.js

```diff
@@ -482,11 +477,106 @@ define([
+                });
+                it('Should start with all buttons disabled', async function () {
+                    const arg = {
+                        node: this.node,
+                        jobId,
+                    };
+                    await this.jobLogViewerInstance.start(arg);
+                    const btns = this.node.querySelectorAll('div[data-element="header"] button');
+                    btns.forEach((btn) => {
+                        expect(btn).toHaveClass('disabled');
+                    });
+                });
+
+                it('should have an expand button that toggles the log container class', function () {
+                    const standardClass = `.${cssBaseClass}__content`,
+                        expandedClass = `${standardClass}--expanded`,
+                        expandButton = this.node.querySelector(
+                            `.${cssBaseClass}__log_button--expand`
+                        );
+
+                    expect(this.node.querySelectorAll(standardClass).length).toEqual(1);
+                    expect(this.node.querySelectorAll(expandedClass).length).toEqual(0);
+
+                    expandButton.click();
+                    expect(this.node.querySelectorAll(standardClass).length).toEqual(0);
+                    expect(this.node.querySelectorAll(expandedClass).length).toEqual(1);
+
+                    expandButton.click();
+                    expect(this.node.querySelectorAll(standardClass).length).toEqual(1);
+                    expect(this.node.querySelectorAll(expandedClass).length).toEqual(0);
+                });
+
+                it('Should have the top button go to the top', function () {
+                    const logContent = this.node.querySelector('.kb-log__content'),
+                        topButton = this.node.querySelector(`.${cssBaseClass}__log_button--top`);
+                    // set the scrollTop to the midway point
+                    logContent.scrollTop = logContent.clientHeight / 2;
+
+                    topButton.click();
+                    expect(logContent.scrollTop).toEqual(0);
+                });
+
+                it('Should have the bottom button go to the end', function () {
+                    const logContent = this.node.querySelector('.kb-log__content'),
+                        bottomButton = this.node.querySelector(
+                            `.${cssBaseClass}__log_button--bottom`
+                        );
+                    // set the scrollTop to the midway point
+                    logContent.scrollTop = logContent.clientHeight / 2;
+
+                    bottomButton.click();
+                    expect(logContent.scrollTop).not.toEqual(0);
+                    expect(logContent.scrollTop).toEqual(
+                        logContent.scrollHeight - logContent.clientHeight
+                    );
+                });
+
+                xit('should have stop and play buttons to turn logs off and on', () => {});
```

Comment:
I started writing tests for these but they did not behave as expected, so leaving them for now